libs/cde-visualization/src/lib/histogram/histogram.component.ts

```diff
@@ -0,0 +1,207 @@
+  templateUrl: './histogram.component.html',
+  styleUrl: './histogram.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HistogramComponent implements AfterViewInit {
+  /** Visualization element */
+  @ViewChild('vis') vis?: ElementRef;
+
+  /** Data */
+  @Input() data: HistogramData[] = TEST_DATA;
+
+  /** Vega lite spec for visualization */
+  spec: VisualizationSpec = {};
+
+  panelOpen = true;
+
+  colorMap: ColorMap = [
+    {
+      cell_type: 'a',
+      cell_color: [189, 189, 189],
+      anchor: true,
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+    {
+      cell_type: 'b',
+      cell_color: [93, 102, 127],
+    },
+  ];
+
+  currentColor = '';
+
+  ngAfterViewInit(): void {
+    this.spec = this.createHistogram(this.data);
+    if (this.vis) {
+      embed(this.vis.nativeElement, this.spec, { actions: false, renderer: 'svg' });
+    }
+  }
+
+  toRGB(color: [number, number, number]): string {
+    return `rgba(${color.join(', ')})`;
+  }
+
+  rgbToHex(color: [number, number, number]) {
+    return '#' + ((1 << 24) + (color[0] << 16) + (color[1] << 8) + color[2]).toString(16).slice(1);
+  }
+
+  // setCurrentColor(color: [number, number, number]) {
+  //   this.currentColor = this.rgbToHex(color);
+  // }
+
+  private createHistogram(data: HistogramData[]): VisualizationSpec {
+    return {
+      $schema: 'https://vega.github.io/schema/vega-lite/v5.json',
+      width: 'container',
+      height: 'container',
+      data: {
+        values: data,
+      },
+      mark: 'line',
+      encoding: {
+        x: {
+          field: 'distance',
+          type: 'quantitative',
+          title: 'Distance (µm)',
+          scale: { zero: true },
+        },
+        y: {
+          field: 'numCells',
+          type: 'quantitative',
+          title: 'Number of Cells',
+        },
+        color: {
+          field: 'type',
+          type: 'nominal',
+          legend: null,
+          scale: { range: this.colorMap.map((entry) => this.rgbToHex(entry.cell_color)) },
+        },
+      },
+    };
+  }
+
+  download(event: MouseEvent, type: 'svg' | 'png') {
+    event.stopPropagation();
+    embed(this.vis?.nativeElement, this.spec).then((result) => {
```

Comment:
You shouldn't have to recreate the visualization here. Save the view in line 145 and use it here instead of using embed again.